src/main/java/net/minecraftforge/entity/attributes/BooleanAttribute.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.entity.attributes;
+
+import net.minecraft.entity.ai.attributes.RangedAttribute;
+
+public class BooleanAttribute extends RangedAttribute
+{
+
+    public BooleanAttribute(String descriptionId, boolean enabledByDefault)
+    {
+        super(descriptionId, enabledByDefault ? 1 : 0, 0, 1);
+    }
+
+    @Override
+    public double sanitizeValue(double valueIn)
+    {
+        return valueIn > 0 ? 1 : 0;
+    }
+
+    @Override
+    public boolean isBooleanAttribute() {
+        return true;
+    }
+
+    /*
+    To reduce headache, every one of these calculations is treated as addition.
```

Comment:
I think the nuance of having multiplicative values should be preserved. Otherwise, you could never add an effect that disables fall flying (or any other boolean attribute) without knowing that, with enough sources given to them, a player could just overcome that behavior. Being able to simply multiply the attribute by 0 would ensure that the intended behavior is carried through.

Reply:
That’s a valid point, and an idea I had for being able to customize how an attribute is displayed in a tooltip would be effective in conveying what exactly is being done by the attribute modifier.

Functionally though, I imagine there would be no significant difference between a MULTIPLY_BASE and MULTIPLY_TOTAL for a Boolean Attribute.

Reply:
Should be resolved with latest commit